### ISSUE TITLE
Lexer: `a = ; 5`: Suppressed newline should be unsuppressed by semicolon

### ISSUE THREAD

**Issue by vendethiel** [jashkenas/coffeescript#3098]
(from #3053)

``` coffee
# rule : 
'Assignable = TERMINATOR Expression'
# supposedly allowing that :
a =
5
# but also allows this
a = ; 5
```

**Comment by GeoffreyBooth** [comment]
I’m assuming the solution for this is to have the semicolon “un-suppress” the newline that gets suppressed by the open `=`.

**Comment by GeoffreyBooth** [comment]
Fixed via #4669.
